src/shogun/machine/gp/KLDualInferenceMethod.cpp

```diff
@@ -126,10 +127,96 @@ void CKLDualInferenceMethod::init()
 	SG_ADD(&m_is_dual_valid, "is_dual_valid",
 		"whether the lambda (m_W) is valid or not",
 		MS_NOT_AVAILABLE);
+
```

Comment:
KLDualInferenceMethod requires  a bound constrained optimizer.
I will send another PR (a build-in optimizer)  to address this.